### PR TITLE
Issue #3032094: Mechanism for updating config structure.

### PR DESCRIPTION
**Pull request by kshamadeshmukh**

### CODE REVIEW

File: src/Service/Core.php [hunk 1]
```diff
@@ -292,4 +292,52 @@ public function getJsonLink($app_id, $revision = NULL) {
...
+    if (empty($config_json)) {
+      throw new \Exception(dt('@module does not provide JSON configuration.', [
+        '@module' => $module,
+      ]));
+    }
+
+    // Get existing config for an app.
+    $app_config = $this->getAppConfig($module);
+
+    // Merge the configs to aviod any overrides for
```
Comment (malcomio) [inline src/Service/Core.php]: typo - aviod

Comment (kshamadeshmukh) [reply src/Service/Core.php]: Fixed.

File: src/Service/Core.php [hunk 2]
```diff
@@ -292,4 +292,52 @@ public function getJsonLink($app_id, $revision = NULL) {
...
+   *   The machine name of the module.
+   * @param array $config_json
+   *   The configuration settings.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  public function setAppConfig($module, array $config_json = NULL) {
+    $config = [];
```
Comment (malcomio) [inline src/Service/Core.php:311]: this variable is unused - declared here, but then overwritten in line 329

File: src/Service/Core.php [hunk 3]
```diff
@@ -292,4 +292,52 @@ public function getJsonLink($app_id, $revision = NULL) {
...
+        '@module' => $module,
+      ]));
+    }
+
+    // Get existing config for an app.
+    $app_config = $this->getAppConfig($module);
+
+    // Merge the configs to avoid any overrides for
+    // the changes done in App config.
+    $config = NestedArray::mergeDeep($config_json, $app_config);
```
Comment (malcomio) [inline src/Service/Core.php:329]: need to import this class

**Comment by malcomio** [comment]
replaced by #43

**Pull request closed without merging**